src/app/shared/components/template/components/progress-path/progress-path.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from "@angular/core";
+import { TemplateBaseComponent } from "../base";
+import { getStringParamFromTemplateRow } from "src/app/shared/utils";
+
+interface IProgressPathParams {
+  /** TEMPLATE PARAMETER: path_segment_asset */
+  pathSegmentAsset: string;
```

Comment:
nit(blocking): given how critical the path segment asset is to rendering the component might make more sense to hardcode the SVG and drop the authoring option. Instead could expose some sort of styling for the path asset (e.g. color). For that you will want to edit the original svg to support currentColor, e.g. `<svg fill="currentColor">`, and then just specify the shade of gray from CSS

Reply:
Could also (in the future) consider different styles of path, e.g.
https://zellwk.com/blog/million-devs-paths/

Reply:
Alternatively you could also consider a pure CSS approach like the ones seen in:
https://codepen.io/j471n/pen/vYJaLvm
https://codepen.io/cplepage/pen/EozVXL

These basically just use elements with rounded borders to set up the path, where each row essentially has whatever combination of left-bend, center straight, and right-bend elements. A few more examples at:
https://freefrontend.com/css-timelines/

Reply:
Thanks, this was helpful. I did consider both of these routes, but originally chose to pass in an SVG asset as I thought this would give me the quickest results. Returning to this, I've chosen to use an inline SVG, which works quite nicely. And I've exposed the style variables (for the colour of the path and the stroke width of the path) at the _theme_ level, rather than at the template component-instantiation level, see [this commit](https://github.com/IDEMSInternational/open-app-builder/pull/2370/commits/fa704a4ab643f1c8fc455f9e58761bf9c3fdb1ca).

The pure CSS approaches are nice and seem to be the best way to make the path line responsive to different content heights (which would be nice to have). But I agree that they're quite hard to work with – leveraging `::before`/`::after` in that way is pretty opaque to me (although I did end up using something similar for the [text-bubble component](https://github.com/IDEMSInternational/open-app-builder/pull/2353/files#diff-ec59163eb361ae21abc556f9688813c90cdf09c47d0733240c345b4eeec68ac2)).